File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.js

function cmp(a, b) {
    return a.rank - b.rank
}
function solution(rank, attendance) {
    const arr = []
    for(let id = 0; id < rank.length; id++) {
        if(attendance[id]) {
            arr.push({
            rank: rank[id],
            id
            })
        }   
    }
    arr.sort(cmp)
    console.log(arr)
    return arr[0].id * 10000 + arr[1].id * 100 + arr[2].id
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.py
def solution(rank, attendance):
    rank = [(i, rank[i]) for i in range(len(rank)) if attendance[i]]
    rank = sorted(rank, key=lambda x: x[1])[:3]
    arr = []
    for x,y in rank:
        arr.append(x)
    a,b,c = arr
    return 10000 * a + 100 * b + c